### PR TITLE
Support running Argo CD/cluster-agent on workload cluster

### PR DESCRIPTION
**Pull request by chetan-rns**
#### Description:
- Update Makefile to install Argo CD and cluster-agent in the workload cluster
- Generate credentials for accessing the workload cluster
- Update APIResourceSchemas for Appstudio CRDs
- Use a different client for GitOpsEngine/workload cluster.

#### Link to JIRA Story (if applicable):

https://issues.redhat.com/browse/GITOPSRVCE-237

### CODE REVIEW

File: backend/eventloop/eventlooptypes/types.go [hunk 1]
```diff
@@ -126,20 +128,78 @@ func GetWorkspaceIDFromNamespaceID(namespace corev1.Namespace) string {
...
+	err = corev1.AddToScheme(scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	k8sClient, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (jgwest) [inline backend/eventloop/eventlooptypes/types.go:157]: Could you add a couple short comments here briefing explaining why we are doing this, e.g. why we are retrieving this secret. Something like "this retrieves the information required to connect to the Argo CD instance, which is stored in the secret."

File: backend/eventloop/eventlooptypes/types.go [hunk 2]
```diff
@@ -126,20 +128,78 @@ func GetWorkspaceIDFromNamespaceID(namespace corev1.Namespace) string {
 	return string(namespace.UID)
 }
 
-func GetK8sClientForGitOpsEngineInstance(gitopsEngineInstance *db.GitopsEngineInstance) (client.Client, error) {
+// GetK8sClientForGitOpsEngineInstance returns a client for accessing resources from a GitOpsEngine cluster based on the environment.
+// KCP environment: return a client that targets a workload cluster where the GitOpsEngine instance is deployed.
+// non-KCP environment: return a normal client that targets the same cluster as backend.
+func GetK8sClientForGitOpsEngineInstance(ctx context.Context, gitopsEngineInstance *db.GitopsEngineInstance) (client.Client, error) {
```
Comment (jgwest) [inline backend/eventloop/eventlooptypes/types.go:134]: Can you add a quick unit test for this function?

The test I'm thinking of this:

If 'IsRunningAgainstKCP' is true, ensure that this function returns an error if the Secret does not exist. Likewise if the Secret does exist, but is in the incorrect format (for example, is missing 'host').

File: kcp/utils.sh [hunk 3]
```diff
@@ -374,4 +333,34 @@ EOF
...
+  name: gitops-engine-cluster-token
+  namespace: gitops
+  annotations:
+    kubernetes.io/service-account.name: gitops-engine-cluster
+type: kubernetes.io/service-account-token
+EOF
+
+  host=$(KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl config view --minify -o jsonpath='{.clusters[0].cluster.server}')
+
+  while ! KUBECONFIG="${WORKLOAD_KUBECONFIG}" kubectl describe secret gitops-engine-cluster-token | grep -E '^token' >/dev/null; do
```
Comment (jgwest) [inline kcp/utils.sh:357]: Missing `-n gitops` here. This line assumes it is the `gitops` namespace already.

File: backend/eventloop/application_event_loop/application_event_runner_deployments.go [hunk 4]
```diff
@@ -701,9 +701,8 @@ func (a applicationEventLoopRunner_Action) cleanOldGitOpsDeploymentEntry(ctx con
 		}
 	}
 
-	// 5) Now that we've deleted the Application row, create the operation that will cause the Argo CD application
-	// to be deleted.
-	gitopsEngineClient, err := a.getK8sClientForGitOpsEngineInstance(gitopsEngineInstance)
+	// Create the operation that will delete the Argo CD application
```
[the remark this answers is not in the document]
Comment (chetan-rns) [reply backend/eventloop/application_event_loop/application_event_runner_deployments.go:704]: Sorry. It wasn't intentional. I guess the comment got updated while fixing merge conflicts. Changing it back.

**Comment by chetan-rns** [comment]
@jgwest I've addressed the review comments. PTAL

**Review by jgwest — APPROVED** [review]
LGTM, thanks again @chetan-rns!

**Pull request merged**